com.unity.sg2/Editor/GraphDeltaRegistry/GraphDelta/GraphHandlerUtils.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.ContextLayeredDataStorage;
+
+namespace UnityEditor.ShaderGraph.GraphDelta
+{
+    static class GraphHandlerUtils
+    {
+        private static void TopoSort(
+            GraphHandler graph,
+            ElementID nodeID,
+            ref HashSet<ElementID> visited,
+            ref List<ElementID> result,
+            in Dictionary<ElementID, HashSet<ElementID>> dependencyList,
+            bool otherContextAreLeaf)
+        {
+            if (!visited.Contains(nodeID))
```

Comment:
I don't think you need this check.

Reply:
The visited list can be reused across multiple calls of `TopoSort(...)`... such as in `98:GetNodesTopologically(...)`, but I'm redundantly checking there as well and this function is private.

Okay yea-- should be fine to remove this one. I'll add a TODO to support a few more variations of usage-- such as (exclude/include root node).